Code/GraphMol/Wrap/props.hpp

```diff
@@ -76,42 +77,67 @@ bool AddToDict(const U &ob, boost::python::dict &dict, const std::string &key) {
 
 template <class T>
 boost::python::dict GetPropsAsDict(const T &obj, bool includePrivate,
-                                   bool includeComputed) {
+                                   bool includeComputed, bool autoConvert=true) {
   boost::python::dict dict;
-  // precedence double, int, unsigned, std::vector<double>,
-  // std::vector<int>, std::vector<unsigned>, string
+  auto &rd_dict = obj.getDict();
+  auto &data = rd_dict.getData();
+  
   STR_VECT keys = obj.getPropList(includePrivate, includeComputed);
-  for (size_t i = 0; i < keys.size(); ++i) {
-    if (AddToDict<int>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<unsigned int>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<bool>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<double>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<std::vector<int>>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<std::vector<unsigned int>>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<std::vector<double>>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<std::vector<std::string>>(obj, dict, keys[i])) {
-      continue;
-    }
-    if (AddToDict<std::string>(obj, dict, keys[i])) {
+  for (auto &rdvalue: data) {
+    if (std::find(keys.begin(), keys.end(), rdvalue.key) == keys.end())
```

Comment:
Out of curiosity: how can it happen that one of the keys from rdvalue is not in `keys`?